browser-sign-in/OktaBrowserSignIn/WelcomeViewController.swift

```diff
@@ -15,89 +15,68 @@
  */
 
 import UIKit
-import OktaOidc
+import WebAuthenticationUI
 
 final class WelcomeViewController: UIViewController {
-
-    var oktaOidc: OktaOidc?
-    var stateManager: OktaOidcStateManager?
+    lazy var auth = WebAuthentication.shared
     
+    @IBOutlet weak var signInButton: UIButton!
     @IBOutlet weak var ephemeralSwitch: UISwitch!
+    @IBOutlet weak var clientIdLabel: UILabel!
     
     override func viewDidLoad() {
         super.viewDidLoad()
         navigationItem.hidesBackButton = true
-        
-        if #available(iOS 13.0, *) {
-            // We'll allow the switch to be shown
+        if let clientId = auth?.signInFlow.client.configuration.clientId {
+            self.clientIdLabel.text = "clientId: \(clientId)"
         } else {
-            ephemeralSwitch.superview?.isHidden = true
-        }
-        
-        do {
-            if let configForUITests = self.configForUITests {
-                oktaOidc = try OktaOidc(configuration: OktaOidcConfig(with: configForUITests))
-            } else {
-                oktaOidc = try OktaOidc()
-            }
-        } catch let error {
-            DispatchQueue.main.async {
-                let alert = UIAlertController(title: "Error", message: error.localizedDescription, preferredStyle: .alert)
-                alert.addAction(UIAlertAction(title: "OK", style: .default, handler: nil))
-                self.present(alert, animated: true, completion: nil)
-            }
-            return
-        }
-        
-        if  let oktaOidc = oktaOidc,
-            let _ = OktaOidcStateManager.readFromSecureStorage(for: oktaOidc.configuration)?.accessToken {
-            self.stateManager = OktaOidcStateManager.readFromSecureStorage(for: oktaOidc.configuration)
-            performSegue(withIdentifier: "show-details", sender: self)
+            self.clientIdLabel.text = "ClientId not configured"
+            self.signInButton.isEnabled = false
         }
     }
-
+    
+    func navigateToDetailsPage() {
+        guard Credential.default != nil else { return }
+        self.performSegue(withIdentifier: "show-details", sender: self)
+    }
+    
+    @IBAction func ephemeralSwitchChanged(_ sender: Any) {
+        guard let sender = sender as? UISwitch else { return }
+        self.auth?.ephemeralSession = sender.isOn
+    }
```

Comment:
There's a bug where the ephemeral session status isn't updated when the view goes in the background.

1. Turn Ephemeral Session On
2. Sign In
3. Sign Out
4. Notice how the Ephemeral Session switch is now in the "Off" position
5. Sign in

The user isn't prompted with a system alert, indicating ephemeral sessions are still enabled.

My suggestion is to remove the ephemeral switch action function altogether. Instead, when the "Sign In" button is tapped, set

```
auth?.ephemeralSession = ephemeralSwitch.isOn
```

Reply:
hmm, good call on this, This way whenever sign in is selected, the ephemeralSession status is the latest value.